tests/test_krylov.py

```diff
@@ -0,0 +1,304 @@
+    0.00000000e00 + 0.0j,
+    0.00000000e00 + 0.0j,
+    0.00000000e00 + 0.0j,
+    0.00000000e00 + 0.0j,
+    0.00000000e00 + 0.0j,
+    0.00000000e00 + 0.0j,
+    0.00000000e00 + 0.0j,
+    0.00000000e00 + 0.0j,
+    1.01806232e-10 + 0.0j,
+    -3.84610666e-10 + 0.0j,
+    0.00000000e00 + 0.0j,
+    -4.86686586e-11 + 0.0j,
+    8.03322966e-11 + 0.0j,
+    0.00000000e00 + 0.0j,
+    -4.86686586e-11 + 0.0j,
+    8.03322966e-11 + 0.0j,
+    0.00000000e00 + 0.0j,
+    1.01806232e-10 + 0.0j,
+    -3.84610666e-10 + 0.0j,
+    0.00000000e00 + 0.0j,
+]
+
+
+def solu_krylov(mapdl, frq):
+    mapdl.allsel()
+    mapdl.run("/SOLU")
+    mapdl.antype("HARMIC")  # HARMONIC ANALYSIS
+    mapdl.hropt("KRYLOV")
+    mapdl.eqslv("SPARSE")
+    mapdl.harfrq(frq)  # Set beginning and ending frequency
+    mapdl.nsubst(1)  # Set the number of frequency increments
+    mapdl.wrfull(1)  # GENERATE .FULL FILE AND STOP
+    mapdl.solve()
+    mapdl.finish()
+
+
+def test_krylov_with_point_load(mapdl):
+    if not meets_version(mapdl._server_version, (0, 5, 0)):
+        pytest.skip("Requires MAPDL v222 and above")
+
+    # Case1 : Run Krylov Pymapdl
+    mapdl.clear()
+    mm = mapdl.math
+    mapdl.jobname = "point_load_py"
+
+    # Parameters set for Krylov
+    max_q = 10
+    frq = 100
+
+    mapdl.cdread("db", os.path.join(lib_path, "krylov_point_load"), "cdb")
+    solu_krylov(mapdl, frq)
+
+    dd = mapdl.krylov
+    dd.krygensub(max_q, frq, True, True).asarray()
+    dd.krysolve(frq, frq, 1, 1, True).asarray()
+    dd.kryexpand(True, 3)
+    Xii_py = mm.vec(name="Xii").asarray()
+
+    # setting the absolute and relative tolerance
+    rtol = 1e-16
+    atol = 1e-16
+
+    # Verify DOF solution
+    assert np.allclose(Xii_macro_pt_load, Xii_py, rtol, atol)
+
+
+@pytest.mark.parametrize("res_key", [0, 1, 2, 3])
+def test_krylov_with_pressure_load(mapdl, res_key):
+    if not meets_version(mapdl._server_version, (0, 5, 0)):
+        pytest.skip("Requires MAPDL v222 and above")
+
+    # With ramped loading
+    # Case1 : Run Krylov Pymapdl
+    mapdl.clear()
+    mm = mapdl.math
+    mapdl.jobname = "pressure_py"
+
+    # Parameters set for Krylov
+    max_q = 10
+    frq = 100
+
+    mapdl.cdread("db", os.path.join(lib_path, "krylov_pressure_load"), "cdb")
+    # Run Harmonic Analysis
+    solu_krylov(mapdl, frq)
+
+    dd = mapdl.krylov
+    dd.krygensub(max_q, frq, True, True).asarray()
+    dd.krysolve(frq, frq, 1, 0, True).asarray()
+    dd.kryexpand(True, res_key)
+    Xii_py = mm.vec(name="Xii").asarray()
+
+    # setting the absolute and relative tolerance
+    rtol = 1e-16
+    atol = 1e-16
+
+    # Verify DOF solution
+    assert np.allclose(Xii_macro_pres_load, Xii_py, rtol, atol)
+
+
+# Test Exceptions
+@pytest.mark.parametrize(
+    "input_kry_gensub",
+    [
+        (
+            10.2,
+            100,
+            True,
+            True,
+            "The maximum size of Krylov subspace is required to be greater than 0",
```

Comment:
I'm not sure if all of the test outputs have to be clear, complete sentences. If so, reformat similar to "The maximum size of the Krylov subspace must be greater than 0." Make similar changes in subsequent messages.

Reply:
I double checked.